docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -146,6 +230,153 @@ Workflow methods annotated with `@QueryMethod` and `@SignalMethod` can be invoke
+- Implement Signals and Queries dynamically.
+
+  ```Java
+        Workflow.registerListener(
+          (DynamicSignalHandler)
+              (signalName, encodedArgs) -> name = encodedArgs.get(0, String.class));
+  ```
+
+- Since `DynamicWorkflow` can be invoked for different Workflow Types, to check what type is running when your Dynamic Workflow `execute` method runs, use:
+
+  ```Java
+  String type = Workflow.getInfo().getWorkflowType();
+  ```
+
+- All standard `WorkflowOptions` apply to Dynamic Workflows.
+- All the determinism rules apply to Workflows that implement this interface.
+
+The following example shows a Dynamic Workflow Implementation.
+
+```Java
+// Dynamic Workflow Implementation
+public static class DynamicGreetingWorkflowImpl implements DynamicWorkflow {
+  private String name;
+
+  @Override
+  public Object execute(EncodedValues args) {
+    String greeting = args.get(0, String.class);
+    String type = Workflow.getInfo().getWorkflowType();
+
+    // Register dynamic Signal handler
+    Workflow.registerListener(
+        (DynamicSignalHandler)
+            (signalName, encodedArgs) -> name = encodedArgs.get(0, String.class));
+```
+
+The following example shows how to register the Dynamic Workflow implementation with a Worker.
+
+```Java
+  public static void main(String[] arg) {
+
+    WorkflowServiceStubs service = WorkflowServiceStubs.newInstance();
+    WorkflowClient client = WorkflowClient.newInstance(service);
+    WorkerFactory factory = WorkerFactory.newInstance(client);
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    /* Register the Dynamic Workflow implementation with the Worker. Workflow implementations
+    ** must be known to the Worker at runtime in order to dispatch Workflow Tasks.
+    */
+    worker.registerWorkflowImplementationTypes(DynamicGreetingWorkflowImpl.class);
+
+    // Start all the Workers that are in this process.
+    factory.start();
+
+    /* Create the Workflow stub. Note that the Workflow type is not explicitly registered with the Worker */
+    WorkflowOptions workflowOptions =
+        WorkflowOptions.newBuilder().setTaskQueue(TASK_QUEUE).setWorkflowId(WORKFLOW_ID).build();
+    WorkflowStub workflow = client.newUntypedWorkflowStub("DynamicWF", workflowOptions);
+
+    /* Start Workflow Execution and Signal right after. Pass in the Workflow args and Signal args */
+    workflow.signalWithStart("greetingSignal", new Object[] {"John"}, new Object[] {"Hello"});
+
+    // Wait for the Workflow to finish getting the results
+    String result = workflow.getResult(String.class);
+
+    System.out.println(result);
+
+    System.exit(0);
+  }
+}
+```
+
 ### Workflow implementation constraints
 
 Temporal uses the [Event Sourcing pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/event-sourcing) to recover the state of a Workflow object including its threads and local variable values.
```

Comment:
Lets get rid of this -
We don't actually use an Event Sourcing pattern.

Reply:
Done.